ReactAndroid/src/main/java/com/facebook/react/animated/DecayAnimation.java

```diff
@@ -44,8 +48,15 @@ public void runAnimationStep(long frameTimeNanos) {
         (1 - Math.exp(-(1 - mDeceleration) * (frameTimeMillis - mStartFrameTimeMillis)));
 
     if (Math.abs(mLastValue - value) < 0.1) {
-      mHasFinished = true;
-      return;
+
+      if (mNumLoops == -1 || mCurrentLoop < mNumLoops) { // looping animation, return to start
+        mStartFrameTimeMillis = -1;
```

Comment:
On line 39 there is a conditional checking whether mStartFrameTimeMillis is -1 (as it is on construction), and if so runs some initialisation code. I'd thought it best to let the animation driver go through the same code path on restarting a loop, rather than duplicating the code in the restart section. Let me know if you still think I should change it.